text/1006-deprecate-action-template-helper.md

```diff
@@ -0,0 +1,272 @@
+
+Before:
+```js
+import Component from '@glimmer/component';
+
+export default class Demo extends Component {
+    actions = {
+        plusOne() {
+           /* ... */ 
+        }
+    }
+}
+```
+or
+```js
+import Component from '@ember/component';
+
+export default class Demo extends Component {
+    actions = {
+        plusOne() {
+           /* ... */ 
+        }
+    }
+}
+```
+or
+```js
+import Component from '@ember/component';
+
+export default Component.extend({
+    actions: {
+        plusOne() {
+           /* ... */ 
+        }
+    }
+})
+```
+
+After:
+```js
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class Demo extends Component {
+    @action
+    plusOne() {
+       /* ... */ 
+    }
+}
+```
+
+Note that `@action` is completely different from `(action)` or `{{action}}` (and is partly a motivator for deprecating `(action)` and `{{action}}`, to reduce ambiguity).
+
+`@action` is binds the `this` on the method to the instance of the class. 
+
+### Scenario: `action` is passed a function reference
+
+Before:
+```hbs
+<SomeComponent @update={{action this.plusOne}} />
+```
+
+After
+
+```hbs
+<SomeComponent @update={{this.plusOne}} />
+```
+
+### Scenario: `action` is passed parameters
+
+Before:
+```hbs
+<SomeComponent @update={{action this.plus 1}} />
+```
+
+After:
+```hbs
+<SomeComponent @update={{fn this.plus 1}} />
+```
+
+### Scenario: `action` is used with `mut` 
+
+Before:
+```hbs
+<SomeComponent @update={{action (mut @value.property}} />
+```
+After:
+```js
+// parent.js
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class SomeComponent extends Component {
+    @action
+    handleUpdate(value) {
+        this.args.property = value; 
+    }
+}
+```
+```hbs
+{{! parent.hbs }}
+<SomeComponent @update={{this.handleUpdate}} />
+```
+
+Related, [Combining function arguments with action functions](https://guides.emberjs.com/release/components/component-state-and-actions/#toc_combining-arguments-and-actions)
+
+### Related: `send`
+
+When removing `(action)` or `{{action}}` with a _string_ name, you'll also need to verify that there are no [`send`](https://api.emberjs.com/ember/5.6/classes/Component/methods/send?anchor=send) calls with that same string.
+
+## How We Teach This
+
+The guides already cover how to invoke functions in the modern way.
+
+Remove: https://api.emberjs.com/ember/5.6/classes/Ember.Templates.helpers/methods/action?anchor=action
+
+## Drawbacks
+
+Older code will stop working once the deprecated code is removed.
+
+## Alternatives
+
+- adding an import so folks can keep using action in gjs.
+  I don't think we should do this because we want to clean up antiquated patterns, rather than encourage their continued existence.
+
+## Unresolved questions
+
+- Could there be a codemod?
+  _Potentially_ for action usage that references `this.properties`. For string actions, it's impossible.
```

Comment:
And at minima the codemod would be able to emit warnings in that case ?

Reply:
yea